### ISSUE TITLE
Warn on usage of negative unary operator combined with a method call on literal

### ISSUE THREAD

**Issue by dbrgn** [Manishearth/rust-clippy#545]
What I mean is this:

```
1.32.abs(): 1
-1.32.abs(): -1
pos.abs(): 1
neg.abs(): 1
-pos.abs(): -1
-neg.abs(): -1
```

This one should raise a warning because it might be unintuitive:

```
-1.32.abs(): -1
```

Less ambiguous would be the following two versions:

```
(-1.32).abs()
-(1.32.abs())
```

See https://github.com/rust-lang/rust/issues/25564

**Comment by mcarton** [comment]
Isn’t that exactly what you propose?
```rust
% cat test.rs
fn main() {
    println!("{}", -1i64.abs());
    println!("{}", -1.0f64.abs());
}
% rustc -L $CLIPPY_PATH -Z extra-plugins=clippy test.rs
test.rs:2:20: 2:31 warning: unary minus has lower precedence than method call. Consider adding parentheses to clarify your intent: -(1i64.abs()), #[warn(precedence)] on by default
test.rs:2     println!("{}", -1i64.abs());
                             ^~~~~~~~~~~
<std macros>:2:25: 2:56 note: in this expansion of format_args!
<std macros>:3:1: 3:54 note: in this expansion of print! (defined in <std macros>)
test.rs:2:5: 2:33 note: in this expansion of println! (defined in <std macros>)
test.rs:2:20: 2:31 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#precedence
test.rs:3:20: 3:33 warning: unary minus has lower precedence than method call. Consider adding parentheses to clarify your intent: -(1.0f64.abs()), #[warn(precedence)] on by default
test.rs:3     println!("{}", -1.0f64.abs());
                             ^~~~~~~~~~~~~
<std macros>:2:25: 2:56 note: in this expansion of format_args!
<std macros>:3:1: 3:54 note: in this expansion of print! (defined in <std macros>)
test.rs:3:5: 3:35 note: in this expansion of println! (defined in <std macros>)
test.rs:3:20: 3:33 help: for further information visit https://github.com/Manishearth/rust-clippy/wiki#precedence
```

**Comment by Manishearth** [comment]
I didn't even know that the precedence lint did method calls. Sweet.

**Comment by dbrgn** [comment]
Ah, I missed that one :) Great to see that it's already implemented.
